pkg/identity/healthcheckd.go

```diff
@@ -46,24 +45,28 @@ func Healthcheckd(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 		Handler: createHealthCheckServiceMux(idConfig.HealthCheckEndpoint),
 	}
 
+	serverDone := make(chan struct{})
```

Comment:
```suggestion
	serverDone := make(chan struct{}, 1)
```
add size, please change other places too